core/internal/featurechecker/features.go

```diff
@@ -0,0 +1,85 @@
+package featurechecker
+
+import (
+	"context"
+	"sync"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/wandb/wandb/core/internal/gql"
+	"github.com/wandb/wandb/core/internal/observability"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+)
+
+// ServerFeaturesCache is responsible for providing the capabilities of a server
+// Features and capabilities are retrieved from the server via a GraphQL query
```

Comment:
nit: punctuate sentences and use a blank line after the summary sentence

however, I think you can just get rid of the second sentence as it's simply an implementation detail:

```
// ServerFeaturesCache queries the capabilities of the W&B backend.
```